### PR TITLE
handshake_flow

### PR DESCRIPTION
**Pull request by Blindspot22**
The handshake flow should include a clear description of how the communication initiation, message exchanges, and finalization of the handshake are handled

Provide a structured overview of how parties initiate, negotiate, and establish trust during communication.

### CODE REVIEW

File: handshake_docs/handshake_flow.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Handshake Flow for DIDComm Mediator-rs
+
+## 1. Communication Initiation
+### Participant Identification:
+- Each mobile agent generates a DID (Decentralized Identifier).
+
```
Comment (francis-pouatcha) [inline handshake_docs/handshake_flow.md:6]: A mobile agent is called an edge agent.

Generally, the generation of a did involves some interaction with a mediator. This is the cloud agent that is responsible for:

1. Receiving messages sent to the edge agent
2. Forwarding messages sent out by the edge agent.

File: handshake_docs/handshake_flow.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+- Agents exchange their DIDs to identify each other. The initial exchange might involve sending a DIDComm message that includes the agent’s DID and a nonce (a random value used for security purposes).
```
Comment (francis-pouatcha) [inline handshake_docs/handshake_flow.md:7]: This exchange occurs in the of an OOB message. Generally in which one edge agent scans the QR code of another edge agent.

File: handshake_docs/handshake_flow.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+### Initial Message:
+- The initiating agent sends a DIDComm handshake message to the mediator. This message contains:
+
+  - **DID of the initiating agent**: Identifies the sender.
+  - **Nonce**: Ensures message freshness and prevents replay attacks.
+  - **Service Information**: May include endpoint URLs, service types, and other metadata relevant to the communication.
+## 2. Message Exchanges
+### Encryption Methods:
+ - Messages are encrypted using DIDComm’s encryption standards, typically using asymmetric encryption (e.g., using public keys derived from the DIDs).
```
Comment (francis-pouatcha) [inline handshake_docs/handshake_flow.md:16]: We also use to refine this in a workshop. Most of messages sent are hybrid.

File: handshake_docs/handshake_flow.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+   - **Payload**: The actual content of the message, encrypted as needed.
+   - **Encryption Details**: Information about the encryption used, such as keys and algorithms.
+### Sequence of Exchanges:
+ - **Mediator Acknowledgment**: The mediator responds with its own DID and a nonce, acknowledging the receipt of the initial message.
+   - **Capability Negotiation**: Both parties may exchange additional messages to negotiate capabilities, such as supported encryption algorithms, message formats, and routing mechanisms.
+   - **Routing Information**: The mediator may provide routing information necessary for forwarding future messages between the agents.
+   - **Verification**: Each party verifies the identity of the other using cryptographic proofs, ensuring the authenticity and integrity of the exchanged DIDs and messages.
+## 3. Finalization
+### Successful Connection:
+ - The handshake is finalized when both parties have successfully exchanged all necessary cryptographic information and have verified each other’s identities.
```
Comment (francis-pouatcha) [inline handshake_docs/handshake_flow.md:29]: I am not sure of the identity verification. If most of the cases we will have no trust authority. So identities are self claimed.  This is why i would leave out the identity verification.

An out of band exchange of the initial contact is trust enough to start.

**Review by francis-pouatcha — CHANGES_REQUESTED** [review]

File: handshake_docs/handshake_flow.md (hunk 1 above)
Comment (Blindspot22) [reply handshake_docs/handshake_flow.md:6]: Okay sir

**Comment by Blindspot22** [comment]
@francis-pouatcha please review this PR

File: handshake_docs/handshake_flow.md
Comment (francis-pouatcha) [inline handshake_docs/handshake_flow.md:1]: move file in a file named docs/handshake.md so that we can have other documents there as well.

**Review by francis-pouatcha — CHANGES_REQUESTED** [review]

File: handshake_docs/handshake_flow.md
Comment (Blindspot22) [reply handshake_docs/handshake_flow.md:1]: Okay sir

**Comment by Blindspot22** [comment]
Seems like @francis-pouatcha  is busy. Please @Hermann-Core and @ndefokou review this ticket and let me know if you see anything wrong or don't understand anything.

**Review by francis-pouatcha — APPROVED** [review]

**Comment by Hermann-Core** [comment]
@Blindspot22  we should consolidate all project-level documentation-related files into the **`docs`** folder for better organization. Specifically, the **`handshakedoc.md`** file should be renamed to **`handshake.md`**. Since the file will be located within the **`docs`** folder, appending "doc" to the filename is redundant.

**Comment by Blindspot22** [comment]
> @Blindspot22 we should consolidate all project-level documentation-related files into the **`docs`** folder for better organization. Specifically, the **`handshakedoc.md`** file should be renamed to **`handshake.md`**. Since the file will be located within the **`docs`** folder, appending "doc" to the filename is redundant.

Yeahhh it's done.

**Pull request merged**